src/event-engine/states/receive_failed.ts

```diff
@@ -58,7 +58,7 @@ ReceiveFailedState.on(restore.type, (context, { payload }) => {
   return HandshakingState.with({
     channels: payload.channels,
     groups: payload.groups,
-    cursor: { timetoken: payload.cursor.timetoken, region: payload.cursor.region || context.cursor.region },
+    cursor: { timetoken: `${payload.cursor.timetoken}`, region: payload.cursor.region || context.cursor.region },
```

Comment:
Is it due to the fact that we provided type for timetoken as `string | number` at [`SubscribeParameters`](https://github.com/pubnub/javascript/blob/35517724e291af51431efe03ee27962d8ca1e84a/src/core/types/api/subscription.ts#L507)?

Reply:
Yes. I didn't wanted to break interface, so wrapped it into string. We actually never get number in subscribe response from the server (OpenAPI specs for timetoken specify `string` type).